site/src/components/WorkspacesTable/WorkspacesTableBody.tsx

```diff
@@ -1,98 +1,68 @@
 import Button from "@material-ui/core/Button"
 import Link from "@material-ui/core/Link"
 import { makeStyles } from "@material-ui/core/styles"
-import TableCell from "@material-ui/core/TableCell"
-import TableRow from "@material-ui/core/TableRow"
 import AddOutlined from "@material-ui/icons/AddOutlined"
-import { ChooseOne, Cond } from "components/Conditionals/ChooseOne"
+import { Workspace } from "api/typesGenerated"
+import { TableEmpty } from "components/TableEmpty/TableEmpty"
 import { FC } from "react"
 import { useTranslation } from "react-i18next"
 import { Link as RouterLink } from "react-router-dom"
-import { workspaceFilterQuery } from "../../util/filters"
-import { WorkspaceItemMachineRef } from "../../xServices/workspaces/workspacesXService"
-import { EmptyState } from "../EmptyState/EmptyState"
 import { TableLoader } from "../TableLoader/TableLoader"
 import { WorkspacesRow } from "./WorkspacesRow"
 
 interface TableBodyProps {
-  isLoading?: boolean
-  workspaceRefs?: WorkspaceItemMachineRef[]
-  filter?: string
-  isNonInitialPage: boolean
+  workspaces?: Workspace[]
+  isUsingFilter: boolean
+  onUpdateWorkspace: (workspace: Workspace) => void
 }
 
 export const WorkspacesTableBody: FC<
   React.PropsWithChildren<TableBodyProps>
-> = ({ isLoading, workspaceRefs, filter, isNonInitialPage }) => {
+> = ({ workspaces, isUsingFilter, onUpdateWorkspace }) => {
   const { t } = useTranslation("workspacesPage")
   const styles = useStyles()
 
+  if (!workspaces) {
+    return <TableLoader />
+  }
+
+  if (workspaces.length === 0) {
+    return isUsingFilter ? (
```

Comment:
Instead of prop drilling several levels for this, can we look for the existence of the `filter` query string in the URL using react-router? It doesn't seem to be present unless explicitly set by the user.

Reply:
We can, but this brings back the "container/page view" discussion. If we do that, we have to change our stories in Storybook to wrap the components with the `<Router />` and pass the proper routes, adding some work. Let's see how painful it is and we can take a better decision, maybe? Curious to know your thoughts.

Reply:
Sounds good to me!